flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerRunner.java

```diff
@@ -357,18 +359,21 @@ public static TaskExecutor startTaskManager(
 				remoteAddress,
 				localCommunicationOnly);
 
+		TaskManagerMetricGroup taskManagerMetricGroup = MetricUtils.instantiateTaskManagerMetricGroup(
+			metricRegistry, TaskManagerLocation.getHostName(remoteAddress), resourceID);
+
 		TaskManagerServices taskManagerServices = TaskManagerServices.fromConfiguration(
 			taskManagerServicesConfiguration,
+			taskManagerMetricGroup,
 			resourceID,
 			rpcService.getExecutor(), // TODO replace this later with some dedicated executor for io.
 			EnvironmentInformation.getSizeOfFreeHeapMemoryWithDefrag(),
 			EnvironmentInformation.getMaxJvmHeapMemory());
 
-		TaskManagerMetricGroup taskManagerMetricGroup = MetricUtils.instantiateTaskManagerMetricGroup(
-			metricRegistry,
-			taskManagerServices.getTaskManagerLocation(),
-			taskManagerServices.getNetworkEnvironment(),
-			taskManagerServicesConfiguration.getSystemResourceMetricsProbingInterval());
+		Optional<Time> systemResourceProbeInterval = ConfigurationUtils.getSystemResourceMetricsProbingInterval(configuration);
```

Comment:
Should `systemResourceProbeInterval` be moved outside of `instantiateTaskManagerMetricGroup`? Could we avoid this change?

Reply:
It is just for keeping the same behavior as before. In previous `instantiateTaskManagerMetricGroup`, it first instantiates network metrics and them instantiates system metrics, so I keep the same sequence.

I would double check whether it is no problem to adjust the sequence. If so we could avoid this change.